### DISCUSSION TITLE
Help to migrate from MySQL to PostgreSQL

### DISCUSSION THREAD

**Discussion by nikiie8** [firefly-iii/firefly-iii#4806] [Firefly III] [answered]
**Description**

Hi all,
I am running Firefly III version 5.5.11. on a self-hosted server Debian 10.0
I am trying to migrate database from MySQL to PostgreSQL.
I made the following steps:
1. Migrate mysql to postgresql with pgloader tool. Check new database. All data store in new db.
2. Update information in .env file about database connection and port number
3. Run recommended commands  
php artisan migrate --seed
php artisan firefly-iii:upgrade-database 

Then all data disappeared from database. I mean all tables exist, but there are no records. 
When I open browser, firefly suggests me to register new user.

Please help what I do wrong? 

Thanks in advance

**Extra info**
 
PostgreSQL - 11.12 version
MySQL - Mariadb 10.14.19

**Comment by JC5** [comment]
Was there anything in the logs? Maybe it's hitting another database?

**Comment by nikiie8** [comment]
Actually there are not many information in logs.
From file ff3-cli

[2021-05-28 09:55:17] local.WARNING: No database version entry is present. Database is assumed to be OLD (version 1).
[2021-05-28 09:56:09] local.WARNING: Database entry exists. Update to latest version (16)
[2021-05-28 10:45:42] local.WARNING: Database entry exists. Update to latest version (16)
[2021-05-28 10:46:06] local.WARNING: Database entry exists. Update to latest version (16)
[2021-05-28 12:27:04] local.ERROR: Exception is: {"class":"Symfony\\Component\\Console\\Exception\\NamespaceNotFoundException","errorMessage":"There are no$
[2021-05-28 12:27:05] local.ERROR: Exception when mailing: Expected response code 250 but got code "530", with message "530 5.7.1 Authentication required
"
[2021-05-28 12:27:05] local.ERROR: There are no commands defined in the "migration" namespace.

Did you mean this?
    migrate {"exception":"[object] (Symfony\\Component\\Console\\Exception\\NamespaceNotFoundException(code: 0): There are no commands defined in the \"mig$

Did you mean this?
    migrate at /var/www/firefly-iii/vendor/symfony/console/Application.php:605)
[stacktrace]
#0 /var/www/firefly-iii/vendor/symfony/console/Application.php(656): Symfony\\Component\\Console\\Application->findNamespace()
#1 /var/www/firefly-iii/vendor/symfony/console/Application.php(254): Symfony\\Component\\Console\\Application->find()
#2 /var/www/firefly-iii/vendor/symfony/console/Application.php(166): Symfony\\Component\\Console\\Application->doRun()
#3 /var/www/firefly-iii/vendor/laravel/framework/src/Illuminate/Console/Application.php(92): Symfony\\Component\\Console\\Application->run()
#4 /var/www/firefly-iii/vendor/laravel/framework/src/Illuminate/Foundation/Console/Kernel.php(129): Illuminate\\Console\\Application->run()
#5 /var/www/firefly-iii/artisan(37): Illuminate\\Foundation\\Console\\Kernel->handle()
#6 {main}
"}
[2021-05-28 13:15:16] local.WARNING: No database version entry is present. Database is assumed to be OLD (version 1).
[2021-05-28 13:15:43] local.WARNING: Database entry exists. Update to latest version (16)
[2021-05-28 13:37:49] local.WARNING: No database version entry is present. Database is assumed to be OLD (version 1).
[2021-05-28 14:36:25] local.WARNING: No database version entry is present. Database is assumed to be OLD (version 1).
[2021-05-28 14:36:42] local.WARNING: Database entry exists. Update to latest version (16)
[2021-05-28 15:05:32] local.WARNING: No database version entry is present. Database is assumed to be OLD (version 1).
[2021-05-28 15:05:39] local.WARNING: Database entry exists. Update to latest version (16)

These are logs from ff3-fpm-fcgi. But as far as I understand these errors refer to the situation when I tried to login without commands php artisan migrate --seed php artisan firefly-iii:upgrade-database

[2021-05-28 15:04:06] local.ERROR: Exception is: {"class":"Illuminate\\Database\\QueryException","errorMessage":"SQLSTATE[42P01]: Undefined table: 7 ERROR:$
[2021-05-28 15:04:06] local.ERROR: Exception when mailing: Expected response code 250 but got code "530", with message "530 5.7.1 Authentication required
"
[2021-05-28 15:04:06] local.ERROR: SQLSTATE[42P01]: Undefined table: 7 ERROR:  relation "users" does not exist
LINE 1: select * from "users" where "id" = $1 limit 1
                      ^ (SQL: select * from "users" where "id" = 1 limit 1) {"exception":"[object] (Illuminate\\Database\\QueryException(code: 42P01): SQLS$
LINE 1: select * from \"users\" where \"id\" = $1 limit 1
                      ^ (SQL: select * from \"users\" where \"id\" = 1 limit 1) at /var/www/firefly-iii/vendor/laravel/framework/src/Illuminate/Database/Co$
[stacktrace]
#0 /var/www/firefly-iii/vendor/laravel/framework/src/Illuminate/Database/Connection.php(638): Illuminate\\Database\\Connection->runQueryCallback()
#1 /var/www/firefly-iii/vendor/laravel/framework/src/Illuminate/Database/Connection.php(346): Illuminate\\Database\\Connection->run()
#2 /var/www/firefly-iii/vendor/laravel/framework/src/Illuminate/Database/Query/Builder.php(2313): Illuminate\\Database\\Connection->select()
#3 /var/www/firefly-iii/vendor/laravel/framework/src/Illuminate/Database/Query/Builder.php(2301): Illuminate\\Database\\Query\\Builder->runSelect()
#4 /var/www/firefly-iii/vendor/laravel/framework/src/Illuminate/Database/Query/Builder.php(2796): Illuminate\\Database\\Query\\Builder->Illuminate\\Databas$
#5 /var/www/firefly-iii/vendor/laravel/framework/src/Illuminate/Database/Query/Builder.php(2302): Illuminate\\Database\\Query\\Builder->onceWithColumns()
#6 /var/www/firefly-iii/vendor/laravel/framework/src/Illuminate/Database/Eloquent/Builder.php(588): Illuminate\\Database\\Query\\Builder->get()
#7 /var/www/firefly-iii/vendor/laravel/framework/src/Illuminate/Database/Eloquent/Builder.php(572): Illuminate\\Database\\Eloquent\\Builder->getModels()
#8 /var/www/firefly-iii/vendor/laravel/framework/src/Illuminate/Database/Concerns/BuildsQueries.php(242): Illuminate\\Database\\Eloquent\\Builder->get()
#9 /var/www/firefly-iii/vendor/laravel/framework/src/Illuminate/Auth/EloquentUserProvider.php(68): Illuminate\\Database\\Eloquent\\Builder->first()
#10 /var/www/firefly-iii/vendor/laravel/framework/src/Illuminate/Auth/SessionGuard.php(179): Illuminate\\Auth\\EloquentUserProvider->retrieveByToken()
#11 /var/www/firefly-iii/vendor/laravel/framework/src/Illuminate/Auth/SessionGuard.php(149): Illuminate\\Auth\\SessionGuard->userFromRecaller()
#12 /var/www/firefly-iii/vendor/laravel/framework/src/Illuminate/Auth/AuthManager.php(54): Illuminate\\Auth\\SessionGuard->user()
#13 [internal function]: Illuminate\\Auth\\AuthManager->Illuminate\\Auth\\{closure}()
#14 /var/www/firefly-iii/vendor/laravel/framework/src/Illuminate/Auth/AuthServiceProvider.php(97): call_user_func()
#15 [internal function]: Illuminate\\Auth\\AuthServiceProvider->Illuminate\\Auth\\{closure}()
#16 /var/www/firefly-iii/vendor/laravel/framework/src/Illuminate/Http/Request.php(518): call_user_func()
17 /var/www/firefly-iii/vendor/laravel/framework/src/Illuminate/Session/Middleware/AuthenticateSession.php(38): Illuminate\\Http\\Request->user()
#18 /var/www/firefly-iii/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(167): Illuminate\\Session\\Middleware\\AuthenticateSession->handle()
#19 /var/www/firefly-iii/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/VerifyCsrfToken.php(78): Illuminate\\Pipeline\\Pipeline->Illumi$
#20 /var/www/firefly-iii/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(167): Illuminate\\Foundation\\Http\\Middleware\\VerifyCsrfToken->han$
#21 /var/www/firefly-iii/vendor/laravel/framework/src/Illuminate/View/Middleware/ShareErrorsFromSession.php(49): Illuminate\\Pipeline\\Pipeline->Illuminate$
#22 /var/www/firefly-iii/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(167): Illuminate\\View\\Middleware\\ShareErrorsFromSession->handle()
#23 /var/www/firefly-iii/vendor/laravel/framework/src/Illuminate/Session/Middleware/StartSession.php(121): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipe$
#24 /var/www/firefly-iii/vendor/laravel/framework/src/Illuminate/Session/Middleware/StartSession.php(64): Illuminate\\Session\\Middleware\\StartSession->ha$
#25 /var/www/firefly-iii/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(167): Illuminate\\Session\\Middleware\\StartSession->handle()
#26 /var/www/firefly-iii/vendor/laravel/framework/src/Illuminate/Cookie/Middleware/AddQueuedCookiesToResponse.php(37): Illuminate\\Pipeline\\Pipeline->Illu$
#27 /var/www/firefly-iii/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(167): Illuminate\\Cookie\\Middleware\\AddQueuedCookiesToResponse->ha$
#28 /var/www/firefly-iii/vendor/laravel/framework/src/Illuminate/Cookie/Middleware/EncryptCookies.php(67): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipe$
#29 /var/www/firefly-iii/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(167): Illuminate\\Cookie\\Middleware\\EncryptCookies->handle()
#30 /var/www/firefly-iii/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(103): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure$
#31 /var/www/firefly-iii/vendor/laravel/framework/src/Illuminate/Routing/Router.php(697): Illuminate\\Pipeline\\Pipeline->then()
#32 /var/www/firefly-iii/vendor/laravel/framework/src/Illuminate/Routing/Router.php(672): Illuminate\\Routing\\Router->runRouteWithinStack()
33 /var/www/firefly-iii/vendor/laravel/framework/src/Illuminate/Routing/Router.php(636): Illuminate\\Routing\\Router->runRoute()
#34 /var/www/firefly-iii/vendor/laravel/framework/src/Illuminate/Routing/Router.php(625): Illuminate\\Routing\\Router->dispatchToRoute()
#35 /var/www/firefly-iii/vendor/laravel/framework/src/Illuminate/Foundation/Http/Kernel.php(166): Illuminate\\Routing\\Router->dispatch()
#36 /var/www/firefly-iii/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(128): Illuminate\\Foundation\\Http\\Kernel->Illuminate\\Foundation\\$
#37 /var/www/firefly-iii/app/Http/Middleware/InstallationId.php(54): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#38 /var/www/firefly-iii/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(167): FireflyIII\\Http\\Middleware\\InstallationId->handle()
#39 /var/www/firefly-iii/vendor/fideloper/proxy/src/TrustProxies.php(57): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#40 /var/www/firefly-iii/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(167): Fideloper\\Proxy\\TrustProxies->handle()
#41 /var/www/firefly-iii/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/TransformsRequest.php(21): Illuminate\\Pipeline\\Pipeline->Illu$
#42 /var/www/firefly-iii/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/ConvertEmptyStringsToNull.php(31): Illuminate\\Foundation\\Http$
#43 /var/www/firefly-iii/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(167): Illuminate\\Foundation\\Http\\Middleware\\ConvertEmptyStringsT$
#44 /var/www/firefly-iii/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/TransformsRequest.php(21): Illuminate\\Pipeline\\Pipeline->Illu$
#45 /var/www/firefly-iii/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/TrimStrings.php(40): Illuminate\\Foundation\\Http\\Middleware\\$
#46 /var/www/firefly-iii/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(167): Illuminate\\Foundation\\Http\\Middleware\\TrimStrings->handle()
#47 /var/www/firefly-iii/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/ValidatePostSize.php(27): Illuminate\\Pipeline\\Pipeline->Illum$
#48 /var/www/firefly-iii/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(167): Illuminate\\Foundation\\Http\\Middleware\\ValidatePostSize->ha$
49 /var/www/firefly-iii/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/PreventRequestsDuringMaintenance.php(86): Illuminate\\Pipeline\$
#50 /var/www/firefly-iii/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(167): Illuminate\\Foundation\\Http\\Middleware\\PreventRequestsDurin$
#51 /var/www/firefly-iii/app/Http/Middleware/SecureHeaders.php(51): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#52 /var/www/firefly-iii/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(167): FireflyIII\\Http\\Middleware\\SecureHeaders->handle()
#53 /var/www/firefly-iii/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(103): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure$
#54 /var/www/firefly-iii/vendor/laravel/framework/src/Illuminate/Foundation/Http/Kernel.php(141): Illuminate\\Pipeline\\Pipeline->then()
#55 /var/www/firefly-iii/vendor/laravel/framework/src/Illuminate/Foundation/Http/Kernel.php(110): Illuminate\\Foundation\\Http\\Kernel->sendRequestThroughR$
#56 /var/www/firefly-iii/public/index.php(76): Illuminate\\Foundation\\Http\\Kernel->handle()
#57 {main}

[previous exception] [object] (PDOException(code: 42P01): SQLSTATE[42P01]: Undefined table: 7 ERROR:  relation \"users\" does not exist
LINE 1: select * from \"users\" where \"id\" = $1 limit 1
                      ^ at /var/www/firefly-iii/vendor/laravel/framework/src/Illuminate/Database/Connection.php:343)
[stacktrace]
#0 /var/www/firefly-iii/vendor/laravel/framework/src/Illuminate/Database/Connection.php(343): PDOStatement->execute()
#1 /var/www/firefly-iii/vendor/laravel/framework/src/Illuminate/Database/Connection.php(671): Illuminate\\Database\\Connection->Illuminate\\Database\\{clos$
#2 /var/www/firefly-iii/vendor/laravel/framework/src/Illuminate/Database/Connection.php(638): Illuminate\\Database\\Connection->runQueryCallback()
#3 /var/www/firefly-iii/vendor/laravel/framework/src/Illuminate/Database/Connection.php(346): Illuminate\\Database\\Connection->run()
#4 /var/www/firefly-iii/vendor/laravel/framework/src/Illuminate/Database/Query/Builder.php(2313): Illuminate\\Database\\Connection->select()
#5 /var/www/firefly-iii/vendor/laravel/framework/src/Illuminate/Database/Query/Builder.php(2301): Illuminate\\Database\\Query\\Builder->runSelect()
#6 /var/www/firefly-iii/vendor/laravel/framework/src/Illuminate/Database/Query/Builder.php(2796): Illuminate\\Database\\Query\\Builder->Illuminate\\Databas$
#7 /var/www/firefly-iii/vendor/laravel/framework/src/Illuminate/Database/Query/Builder.php(2302): Illuminate\\Database\\Query\\Builder->onceWithColumns()
#8 /var/www/firefly-iii/vendor/laravel/framework/src/Illuminate/Database/Eloquent/Builder.php(588): Illuminate\\Database\\Query\\Builder->get()
#9 /var/www/firefly-iii/vendor/laravel/framework/src/Illuminate/Database/Eloquent/Builder.php(572): Illuminate\\Database\\Eloquent\\Builder->getModels()
#10 /var/www/firefly-iii/vendor/laravel/framework/src/Illuminate/Database/Concerns/BuildsQueries.php(242): Illuminate\\Database\\Eloquent\\Builder->get()
#11 /var/www/firefly-iii/vendor/laravel/framework/src/Illuminate/Auth/EloquentUserProvider.php(68): Illuminate\\Database\\Eloquent\\Builder->first()
#12 /var/www/firefly-iii/vendor/laravel/framework/src/Illuminate/Auth/SessionGuard.php(179): Illuminate\\Auth\\EloquentUserProvider->retrieveByToken()
#13 /var/www/firefly-iii/vendor/laravel/framework/src/Illuminate/Auth/SessionGuard.php(149): Illuminate\\Auth\\SessionGuard->userFromRecaller()
#14 /var/www/firefly-iii/vendor/laravel/framework/src/Illuminate/Auth/AuthManager.php(54): Illuminate\\Auth\\SessionGuard->user()
#15 [internal function]: Illuminate\\Auth\\AuthManager->Illuminate\\Auth\\{closure}()
#16 /var/www/firefly-iii/vendor/laravel/framework/src/Illuminate/Auth/AuthServiceProvider.php(97): call_user_func()
#17 [internal function]: Illuminate\\Auth\\AuthServiceProvider->Illuminate\\Auth\\{closure}()
18 /var/www/firefly-iii/vendor/laravel/framework/src/Illuminate/Http/Request.php(518): call_user_func()
#19 /var/www/firefly-iii/vendor/laravel/framework/src/Illuminate/Session/Middleware/AuthenticateSession.php(38): Illuminate\\Http\\Request->user()
#20 /var/www/firefly-iii/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(167): Illuminate\\Session\\Middleware\\AuthenticateSession->handle()
#21 /var/www/firefly-iii/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/VerifyCsrfToken.php(78): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure$
#22 /var/www/firefly-iii/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(167): Illuminate\\Foundation\\Http\\Middleware\\VerifyCsrfToken->handle()
#23 /var/www/firefly-iii/vendor/laravel/framework/src/Illuminate/View/Middleware/ShareErrorsFromSession.php(49): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#24 /var/www/firefly-iii/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(167): Illuminate\\View\\Middleware\\ShareErrorsFromSession->handle()
#25 /var/www/firefly-iii/vendor/laravel/framework/src/Illuminate/Session/Middleware/StartSession.php(121): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#26 /var/www/firefly-iii/vendor/laravel/framework/src/Illuminate/Session/Middleware/StartSession.php(64): Illuminate\\Session\\Middleware\\StartSession->handleStatefulRequest()
#27 /var/www/firefly-iii/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(167): Illuminate\\Session\\Middleware\\StartSession->handle()
#28 /var/www/firefly-iii/vendor/laravel/framework/src/Illuminate/Cookie/Middleware/AddQueuedCookiesToResponse.php(37): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closu$
#29 /var/www/firefly-iii/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(167): Illuminate\\Cookie\\Middleware\\AddQueuedCookiesToResponse->handle()
#30 /var/www/firefly-iii/vendor/laravel/framework/src/Illuminate/Cookie/Middleware/EncryptCookies.php(67): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#31 /var/www/firefly-iii/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(167): Illuminate\\Cookie\\Middleware\\EncryptCookies->handle()
#32 /var/www/firefly-iii/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(103): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#33 /var/www/firefly-iii/vendor/laravel/framework/src/Illuminate/Routing/Router.php(697): Illuminate\\Pipeline\\Pipeline->then()
#34 /var/www/firefly-iii/vendor/laravel/framework/src/Illuminate/Routing/Router.php(672): Illuminate\\Routing\\Router->runRouteWithinStack()
#35 /var/www/firefly-iii/vendor/laravel/framework/src/Illuminate/Routing/Router.php(636): Illuminate\\Routing\\Router->runRoute()
#36 /var/www/firefly-iii/vendor/laravel/framework/src/Illuminate/Routing/Router.php(625): Illuminate\\Routing\\Router->dispatchToRoute()
#37 /var/www/firefly-iii/vendor/laravel/framework/src/Illuminate/Foundation/Http/Kernel.php(166): Illuminate\\Routing\\Router->dispatch()
#38 /var/www/firefly-iii/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(128): Illuminate\\Foundation\\Http\\Kernel->Illuminate\\Foundation\\Http\\{closure}()
#39 /var/www/firefly-iii/app/Http/Middleware/InstallationId.php(54): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#40 /var/www/firefly-iii/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(167): FireflyIII\\Http\\Middleware\\InstallationId->handle()
#41 /var/www/firefly-iii/vendor/fideloper/proxy/src/TrustProxies.php(57): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#42 /var/www/firefly-iii/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(167): Fideloper\\Proxy\\TrustProxies->handle()
#43 /var/www/firefly-iii/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/TransformsRequest.php(21): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closu$
#44 /var/www/firefly-iii/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/ConvertEmptyStringsToNull.php(31): Illuminate\\Foundation\\Http\\Middleware\\Transforms$
#45 /var/www/firefly-iii/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(167): Illuminate\\Foundation\\Http\\Middleware\\ConvertEmptyStringsToNull->handle()
#46 /var/www/firefly-iii/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/TransformsRequest.php(21): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closu$
#47 /var/www/firefly-iii/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/TrimStrings.php(40): Illuminate\\Foundation\\Http\\Middleware\\TransformsRequest->handl$
#48 /var/www/firefly-iii/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(167): Illuminate\\Foundation\\Http\\Middleware\\TrimStrings->handle()
#49 /var/www/firefly-iii/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/ValidatePostSize.php(27): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline
50 /var/www/firefly-iii/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(167): Illuminate\\Foundation\\Http\\Middleware\\ValidatePostSize->handle()
#51 /var/www/firefly-iii/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/PreventRequestsDuringMaintenance.php(86): Illuminate\\Pipeline\\Pipeline->Illuminate\\P$
#52 /var/www/firefly-iii/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(167): Illuminate\\Foundation\\Http\\Middleware\\PreventRequestsDuringMaintenance->handle()
#53 /var/www/firefly-iii/app/Http/Middleware/SecureHeaders.php(51): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#54 /var/www/firefly-iii/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(167): FireflyIII\\Http\\Middleware\\SecureHeaders->handle()
#55 /var/www/firefly-iii/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(103): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#56 /var/www/firefly-iii/vendor/laravel/framework/src/Illuminate/Foundation/Http/Kernel.php(141): Illuminate\\Pipeline\\Pipeline->then()
#57 /var/www/firefly-iii/vendor/laravel/framework/src/Illuminate/Foundation/Http/Kernel.php(110): Illuminate\\Foundation\\Http\\Kernel->sendRequestThroughRouter()
#58 /var/www/firefly-iii/public/index.php(76): Illuminate\\Foundation\\Http\\Kernel->handle()
#59 {main}
"}

**Comment by nikiie8** [comment, marked as answer]
I managed to resolve the issue.
When pgloader imports data from msyql to postgresql it creates an additional schema named as mysql database. While firefly requires that schema should be named as 'public'.
As soon as I fixed, everything worked like a charm.

**Comment by viharm** [comment]
May I ask why you needed to migrate? I am thinking about migrating from _PostgreSQL_ to _MariaDB_ due to the case-insensitive search in the latter. So I want to check if there are any other issues you may have faced in using _MariaDB_/_MySQL_

**Reply by thearchivalone** [reply]
Slightly off topic, but I migrated due to Postgres just being in higher demand in my area. I’m kinda glad I did for two reasons: 1) Postgres’s tab completion out of the box is quite a bit better leading to less typos 2) I haven’t found too many apps I want to use that don’t flat out support postgres these days. Wordpress be damned for saying it’d be too difficult to support in the future
